cmd/azureBlobUpload_test.go

```diff
@@ -37,80 +37,104 @@ func uploadFuncMock(ctx context.Context, api *azblob.BlockBlobClient, file *os.F
+	t.Run("bad tests", func(t *testing.T) {
 		t.Parallel()
-		// create temporary file
-		f, err := os.CreateTemp("", "tmpfile-")
-		if err != nil {
-			log.Fatal(err)
-		}
-		defer f.Close()
-		defer os.Remove(f.Name())
-		data := []byte("test test test")
-		if _, err := f.Write(data); err != nil {
-			log.Fatal(err)
-		}
-		// initialization
-		config := azureBlobUploadOptions{
-			FilePath: f.Name(),
-		}
-		container := mockAzureContainerClient
-		// test
-		err = executeUpload(&config, container(t, true), uploadFuncMock)
-		// assert
-		assert.EqualError(t, err, "Could not instantiate Azure blockBlobClient from Azure Container Client: error containerClient")
-	})
 
-	t.Run("error credentials", func(t *testing.T) {
-		t.Parallel()
-		// initialization
-		config := azureBlobUploadOptions{
-			JSONCredentialsAzure: `{
-				"account_name": "name",
-				"container_name": "container"
-			  }`,
-			FilePath: "nonExistingFilepath",
-		}
-		// test
-		_, err := setup(&config)
-		// assert
-		assert.EqualError(t, err, "Azure credentials are not valid: Key: 'AzureCredentials.SASToken' Error:Field validation for 'SASToken' failed on the 'required' tag")
+		t.Run("error path", func(t *testing.T) {
+			t.Parallel()
+			// initialization
+			config := azureBlobUploadOptions{
+				FilePath: "nonExistingFilepath",
+			}
+			container := mockAzureContainerClient(t, false)
+			// test
+			err := executeUpload(&config, container, uploadFuncMock)
+			// assert
+			assert.IsType(t, &fs.PathError{}, errors.Unwrap(err))
+		})
+
+		t.Run("error containerClient", func(t *testing.T) {
+			t.Parallel()
+			// create temporary file
+			f, err := os.CreateTemp("", "tmpfile-")
+			if err != nil {
+				log.Fatal(err)
+			}
+			defer f.Close()
+			defer os.Remove(f.Name())
+			data := []byte("test test test")
+			if _, err := f.Write(data); err != nil {
+				log.Fatal(err)
+			}
+			// initialization
+			config := azureBlobUploadOptions{
+				FilePath: f.Name(),
+			}
+			container := mockAzureContainerClient(t, true)
+			// test
+			err = executeUpload(&config, container, uploadFuncMock)
+			// assert
+			assert.EqualError(t, err, "Could not instantiate Azure blockBlobClient from Azure Container Client: error containerClient")
+		})
+
+		t.Run("error credentials", func(t *testing.T) {
+			t.Parallel()
+			// initialization
+			config := azureBlobUploadOptions{
+				JSONCredentialsAzure: `{
+					"account_name": "name",
+					"container_name": "container"
+				  }`,
+				FilePath: "nonExistingFilepath",
+			}
+			// test
+			_, err := setup(&config)
+			// assert
+			assert.EqualError(t, err, "Azure credentials are not valid: Key: 'azureCredentials.SASToken' Error:Field validation for 'SASToken' failed on the 'required' tag")
+		})
+
+		t.Run("error JSONStruct", func(t *testing.T) {
+			t.Parallel()
+			// initialization
+			config := azureBlobUploadOptions{
+				JSONCredentialsAzure: `{
+					"account_name": "name",
+					"container_name": "container",
+				  }`,
```

Comment:
Could you make it more obviously wrong?
```suggestion
				JSONCredentialsAzure: `faulty json`,
```